### DISCUSSION TITLE
Running CNN with Quartus backend fails for both, io_stream and io_parallel

### DISCUSSION THREAD

**Discussion by yabezsh** [fastmachinelearning/hls4ml#981] [Q&A]
Dear all,

I am trying to convert a relatively simple custom CNN model to run on fpga with hls4ml using Quartus backend. The problem is that when I use io_stream IOType, I get an error from the compiler: Compiler Error: Multiple reflexive accesses from stream 'layer2_out' is not allowed. I also tried using the model from tests (https://github.com/fastmachinelearning/hls4ml/blob/main/test/pytest/test_cnn_mnist.py) and got the same error message.
I have seen the presentation where it is mentioned that exactly the same issue was observed, but the changes were not applied due to the transition from Intel HLS to oneAPI (https://indico.cern.ch/event/1283970/contributions/5550643/attachments/2721973/4729145/READS%20FastML%20v3.pdf).
 In this thread (https://github.com/fastmachinelearning/hls4ml/pull/656), I also found a suggestion by @jmitrevs on how to change the zeropad1d_cl function to fix the issue. I tried changing the function in templates/quartus/firmware/nnet_utils/nnet_padding_stream.h, but that did not fix the problem. Could I ask for advice on what else should be changed and how (zeropad2d_cl? anything else? any other file?) or if there is any brunch where I can pull/check the working version? 

thanks for any help!

**Comment by jmitrevs** [comment]
If you are still having issues, there's a new oneAPI backend in development. You can try using that. See #955 .
